### ISSUE TITLE
Starting co-ordinates for kernel TICA metadynamics

### ISSUE THREAD

**Issue by sbhakat** [msultan/tica_metadynamics_paper_1#3]
Is there any chance of getting these PDBs as mentioned in the plumed file so that one try to reproduce the exact work with similar settings?

RMSD REFERENCE=../pdbs/0.pdb TYPE=OPTIMAL LABEL=RMSD_0 
RMSD REFERENCE=../pdbs/1.pdb TYPE=OPTIMAL LABEL=RMSD_1 
RMSD REFERENCE=../pdbs/2.pdb TYPE=OPTIMAL LABEL=RMSD_2 
RMSD REFERENCE=../pdbs/3.pdb TYPE=OPTIMAL LABEL=RMSD_3 
RMSD REFERENCE=../pdbs/4.pdb TYPE=OPTIMAL LABEL=RMSD_4 
RMSD REFERENCE=../pdbs/5.pdb TYPE=OPTIMAL LABEL=RMSD_5 
RMSD REFERENCE=../pdbs/6.pdb TYPE=OPTIMAL LABEL=RMSD_6 
RMSD REFERENCE=../pdbs/7.pdb TYPE=OPTIMAL LABEL=RMSD_7 
RMSD REFERENCE=../pdbs/8.pdb TYPE=OPTIMAL LABEL=RMSD_8 
RMSD REFERENCE=../pdbs/9.pdb TYPE=OPTIMAL LABEL=RMSD_9 
RMSD REFERENCE=../pdbs/10.pdb TYPE=OPTIMAL LABEL=RMSD_10 
RMSD REFERENCE=../pdbs/11.pdb TYPE=OPTIMAL LABEL=RMSD_11 
RMSD REFERENCE=../pdbs/12.pdb TYPE=OPTIMAL LABEL=RMSD_12 
RMSD REFERENCE=../pdbs/13.pdb TYPE=OPTIMAL LABEL=RMSD_13 
RMSD REFERENCE=../pdbs/14.pdb TYPE=OPTIMAL LABEL=RMSD_14 
RMSD REFERENCE=../pdbs/15.pdb TYPE=OPTIMAL LABEL=RMSD_15 
RMSD REFERENCE=../pdbs/16.pdb TYPE=OPTIMAL LABEL=RMSD_16 
RMSD REFERENCE=../pdbs/17.pdb TYPE=OPTIMAL LABEL=RMSD_17 
RMSD REFERENCE=../pdbs/18.pdb TYPE=OPTIMAL LABEL=RMSD_18 
RMSD REFERENCE=../pdbs/19.pdb TYPE=OPTIMAL LABEL=RMSD_19 
RMSD REFERENCE=../pdbs/20.pdb TYPE=OPTIMAL LABEL=RMSD_20 
RMSD REFERENCE=../pdbs/21.pdb TYPE=OPTIMAL LABEL=RMSD_21 
RMSD REFERENCE=../pdbs/22.pdb TYPE=OPTIMAL LABEL=RMSD_22 
RMSD REFERENCE=../pdbs/23.pdb TYPE=OPTIMAL LABEL=RMSD_23 
RMSD REFERENCE=../pdbs/24.pdb TYPE=OPTIMAL LABEL=RMSD_24 
RMSD REFERENCE=../pdbs/25.pdb TYPE=OPTIMAL LABEL=RMSD_25 
RMSD REFERENCE=../pdbs/26.pdb TYPE=OPTIMAL LABEL=RMSD_26 
RMSD REFERENCE=../pdbs/27.pdb TYPE=OPTIMAL LABEL=RMSD_27 
RMSD REFERENCE=../pdbs/28.pdb TYPE=OPTIMAL LABEL=RMSD_28 
RMSD REFERENCE=../pdbs/29.pdb TYPE=OPTIMAL LABEL=RMSD_29 
RMSD REFERENCE=../pdbs/30.pdb TYPE=OPTIMAL LABEL=RMSD_30 
RMSD REFERENCE=../pdbs/31.pdb TYPE=OPTIMAL LABEL=RMSD_31 
RMSD REFERENCE=../pdbs/32.pdb TYPE=OPTIMAL LABEL=RMSD_32 
RMSD REFERENCE=../pdbs/33.pdb TYPE=OPTIMAL LABEL=RMSD_33 
RMSD REFERENCE=../pdbs/34.pdb TYPE=OPTIMAL LABEL=RMSD_34 
RMSD REFERENCE=../pdbs/35.pdb TYPE=OPTIMAL LABEL=RMSD_35 
RMSD REFERENCE=../pdbs/36.pdb TYPE=OPTIMAL LABEL=RMSD_36 
RMSD REFERENCE=../pdbs/37.pdb TYPE=OPTIMAL LABEL=RMSD_37 
RMSD REFERENCE=../pdbs/38.pdb TYPE=OPTIMAL LABEL=RMSD_38 
RMSD REFERENCE=../pdbs/39.pdb TYPE=OPTIMAL LABEL=RMSD_39 
RMSD REFERENCE=../pdbs/40.pdb TYPE=OPTIMAL LABEL=RMSD_40 
RMSD REFERENCE=../pdbs/41.pdb TYPE=OPTIMAL LABEL=RMSD_41 
RMSD REFERENCE=../pdbs/42.pdb TYPE=OPTIMAL LABEL=RMSD_42 
RMSD REFERENCE=../pdbs/43.pdb TYPE=OPTIMAL LABEL=RMSD_43 
RMSD REFERENCE=../pdbs/44.pdb TYPE=OPTIMAL LABEL=RMSD_44 
RMSD REFERENCE=../pdbs/45.pdb TYPE=OPTIMAL LABEL=RMSD_45 
RMSD REFERENCE=../pdbs/46.pdb TYPE=OPTIMAL LABEL=RMSD_46 
RMSD REFERENCE=../pdbs/47.pdb TYPE=OPTIMAL LABEL=RMSD_47 
RMSD REFERENCE=../pdbs/48.pdb TYPE=OPTIMAL LABEL=RMSD_48 
RMSD REFERENCE=../pdbs/49.pdb TYPE=OPTIMAL LABEL=RMSD_49

**Comment by msultan** [comment]
sorry, those are not my pdbs but are under the copyright of D.E.Shaw research so I can't hand them out. However, you can email them and get the data yourself and train a new model! They are really nice about giving that data out to the general public.

**Comment by sbhakat** [comment]
I have their .mae file and .dcd files for BPTI which they usually give out. I never saw PDBs with name 0.pdb...49.pdb so on in DESRES hand outs. So I guess these PDBs correspond to some dumped snapshots which in theory should be easily distributable and not within their copyright limit. 

But anyway can you please point out the exact procedure to train the data. I only found .pkl files in this repository but what is there any .py or notebooks which gives the exact procedure.

Thanks for all the help.

**Comment by msultan** [comment]
As it says in the top level Read me, i used the following indices to make the landmark pdbs

For BPTI, we are unable to provide the raw trajectories, featurizer objects, and landmark pdbs due to copy right issues. We note that we used the following 0-indexed locations from 1 ms trajectory(sub-sampled to 25ns) to generate the model. [22337, 37384, 21079, 18818, 34032, 8046, 33230, 8768, 33659, 21386, 40896, 6583, 24395, 1703, 33804, 24607, 8404, 33517, 31602, 32847, 32165, 33077, 14153, 14657, 37158, 38741, 32583, 23406, 35820, 28077, 28222, 17046, 2831, 9374, 33830, 7085, 36247, 29170, 32350, 33175, 18714, 32962, 25604, 33886, 37648, 22345, 28227, 5104, 37812, 32785]


The MSMBuilder KernalTICA module takes in an mdtraj object which should have that. I believe I used all atom RMSD as the distance metric. I am not sure if i have the original script since I will have to check if I still have access to my academic account on our compute cluster.

**Comment by msultan** [comment]
This should roughly (emphasis on roughly) work in generating a tica model.  
```python
  landmark_inds = [22337, 37384, 21079, 18818, 34032, 8046, 33230, 8768, 33659, 21386, 40896, 6583, 24395, 1703, 33804, 24607, 8404, 33517, 31602, 32847, 32165, 33077, 14153, 14657, 37158, 38741, 32583, 23406, 35820, 28077, 28222, 17046, 2831, 9374, 33830, 7085, 36247, 29170, 32350, 33175, 18714, 32962, 25604, 33886, 37648, 22345, 28227, 5104, 37812, 32785]
   shaw_traj = md.load("traj.dcd",top.pdb")
  shaw_landmarks = shaw_traj[landmark_inds]
  for i in range(50):
     shaw_landmarks[i].save_pdb("%d.pdb"%i)

    ktica = KernelTICA(lag_time=5, landmarks= shaw_landmarks)

    y1 = ktica.fit_transform([shaw_traj])[0]
```

Fyi, I am fairly sure you will have to modify the saved pdb files so that Plumed can read them and know which atom indices to use for doing the RMSD calculation. IIRC its something to do with one of the columns have a 1 or a 0 depending on the index.

**Comment by msultan** [comment]
closing now.
